xrpl/models/transactions/did_set.py

```diff
@@ -0,0 +1,49 @@
+"""Model for DIDSet transaction type."""
+
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Pattern
+
+from typing_extensions import Final
+
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import require_kwargs_on_init
+
+HEX_REGEX: Final[Pattern[str]] = re.compile("[a-fA-F0-9]+")
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class DIDSet(Transaction):
+    """Represents a DIDSet transaction."""
+
+    did_document: Optional[str] = None
+    data: Optional[str] = None
+    uri: Optional[str] = None
```

Comment:
Does at least one of these need to be specified? (If so, it doesn't seem like this is enforced in `_get_errors`)

Reply:
Also I don't see a test for that if so

Reply:
Good catch! 🙂